### PR TITLE
[1LP][RFR] Categories, Tags collection conversion

### PR DESCRIPTION
**Pull request by rlbabyuk**
Purpose or Intent
=================
Convert Categories and Tags to collections

{{pytest: -vvv -k 'test_generic_objects_tag_ui or test_paginator_details_page or test_tag_crud or test_category_crud or tagged_vm or test_user_remove_tag'}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit e545718384a523e2f5b72a0db54768686ae8791c:

`cfme/configure/configuration/region_settings.py`:
- :red_circle: Line 86:29: **F821** *undefined name 'TagsCollection'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit bbaf3b4e53cedf07c23dd148c313ba14d5608757:

`cfme/configure/configuration/region_settings.py`:
- :bangbang: Line 137:9: **E265** *block comment should start with '# '*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/configure/configuration/region_settings.py [hunk 1]
```diff
@@ -18,6 +18,112 @@
...
+        return (
+            self.company_tags.is_active() and
+            self.table.is_displayed
+        )
+
+
+@attr.s
+class Tag(Pretty, BaseEntity, Updateable):
+    """ Class represents a category in CFME UI
+        Args:
```
Comment (quarckster) [inline cfme/configure/configuration/region_settings.py]: **REQUIRED**
There should be a blank line after the summary.

Comment (rlbabyuk) [reply cfme/configure/configuration/region_settings.py]: corrected

**Pull request merged**